pkg/integrations/v4/supervisor_fb.go

```diff
@@ -144,6 +164,77 @@ func saveToTempFile(config []byte) (string, error) {
 	return file.Name(), nil
 }
 
+// keeps the most recent config files, up to maxNumberOfFbConfigTempFiles, and removes the rest
+func removeFbConfigTempFiles(maxNumberOfFbConfigTempFiles int) ([]string, error) {
+
+	files, err := os.ReadDir(os.TempDir())
+	if err != nil {
+		return nil, err
+	}
+
+	var fbConfigTempFiles []fs.DirEntry
+
+	for _, file := range files {
+		if !file.IsDir() && strings.HasPrefix(file.Name(), "nr_fb_config") {
+			fbConfigTempFiles = append(fbConfigTempFiles, file)
+		}
+	}
+
+	if len(fbConfigTempFiles) > maxNumberOfFbConfigTempFiles {
+
+		var removedConfigTempFilenames []string
+
+		// sort fbConfigTempFiles by ascending modification date
+		sort.Slice(fbConfigTempFiles, func(i, j int) bool {
+			fileInfo1, _ := fbConfigTempFiles[i].Info()
+			fileInfo2, _ := fbConfigTempFiles[j].Info()
+			return fileInfo1.ModTime().Before(fileInfo2.ModTime())
+		})
+
+		for i := 0; i < len(fbConfigTempFiles)-maxNumberOfFbConfigTempFiles; i++ {
+			fbLuaFilterTempFilenames, err := extractLuaFilterFilenames(fbConfigTempFiles[i].Name())
+
+			if err != nil {
+				return removedConfigTempFilenames, err
+			}
+
+			for _, fbLuaFilterTempFilename := range fbLuaFilterTempFilenames {
+				if err := os.Remove(filepath.Join(os.TempDir(), fbLuaFilterTempFilename)); err != nil {
```

Comment:
We need to be really careful when deleting files in general, but in this case especially as we are deleting files based on the content of a file.
Any user can write to `tmp` folder, create a file called `nr_fb_config_????` and it could add a "malicious" lua line and delete any file in the system as the agent runs normally as root.

I quickly tested it locally and added a test to verify this:

Create a local file `/Users/rruizdegauna/DELETEME` and add it to the `TestRemoveFbConfigTempFiles` test cases :

```
{"nr_fb_lua_filter4", "nr_fb_lua_filter1/../Users/rruizdegauna/DELETEME"},
```

The file is not present after running the tests. 

I might be missing something but let's review this part carefully

Reply:
Ok, I was wrong, the regular expression doesn't catch the `..` Still, let's be extra careful when deleting files 🙏

Reply:
Very good point! I hadn't though about that possibility, but, as you said, the regular expression wouldn't capture `..`. Anyways, I'll take into account this kind of scenery in future changes